### PR TITLE
Parse public key PubKey type incorrectly initialized 

### PR DESCRIPTION
**Pull request by channa-figure**
## Description

Metadata module parsePublicKey was not correctly initializing the PubKey type causing a panic when converting base64 key to pubkey and address.

closes: #225 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by iramiller — APPROVED** [review]
LGTM.  Nice to have test coverage over pieces like this.

File: CHANGELOG.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * Update Cosmos SDK to 0.42.3
 * Remove deprecated ModuleCdc amino encoding from name module #189
 * Remove deprecated ModuleCdc amino encoding from attribute module #188
+* Metadata Module parsing of base64 public key fixed #225
```
Comment (dwedul-figure) [inline CHANGELOG.md:53]: This is in the wrong section. It should be 10 lines up under `## Unreleased`.

Comment (channa-figure) [reply CHANGELOG.md]: Nice catch!

**Review by iramiller — APPROVED** [review]

**Review by dwedul-figure — APPROVED** [review]
LGTM

**Pull request merged**